src/lib/registration/sw_mgmt.rb

```diff
@@ -121,12 +122,15 @@ def self.find_base_product
 
       # use the selected product if a product has been already selected
       selected = product_selected? if Stage.initial
+      selected = AutoinstConfig.selected_product if Mode.auto
```

Comment:
Please use a different variable, `selected` is a Boolean containing a flag whether a product has been already selected (see `self.product_selected?` below). Using a different type (String) in the AY mode is fragile and makes the code hard to read.

Use e.g. `ay_product_name` or something like that.

```ruby
if Mode.auto
  Yast.import "AutoinstConfig"
  ay_product_name = AutoinstConfig.selected_product
end